extension/servicetokenauthextension/factory.go

```diff
@@ -2,36 +2,43 @@ package servicetokenauthextension
 
 import (
 	"context"
+	"errors"
 	"time"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/extension"
 )
 
 var (
+	//nolint:gochecknoglobals
 	Type = component.MustNewType("service_token_auth")
 )
 
 func NewFactory() extension.Factory {
 	return extension.NewFactory(
 		Type,
-		createDefaultConfig,
-		createExtension,
+		CreateDefaultConfig,
+		CreateExtension,
 		component.StabilityLevelAlpha,
 	)
 }
 
-func createDefaultConfig() component.Config {
+func CreateDefaultConfig() component.Config {
```

Comment:
Why this and the one below are exported?

Reply:
Because of the usual suspect `testpackage` :+1: